workflows/mcmicro.nf

```diff
@@ -92,16 +135,26 @@ workflow MCMICRO {
             it.collect{ _1, _2, marker_name, _4, _5, _6 -> '"' + marker_name + '"' }
         }
         .collectFile(name: 'markers.csv', sort: false, newLine: true)
-    ASHLAR.out.tif
-        .join(DEEPCELL_MESMER.out.mask)
+
+    ch_mcquant_masks_markers_key = channel.empty()
```

Comment:
I was thinking this was a lot of code to do something simple, and started reviewing the channel operators. I realized that `cross` accepts a closure to define the mapping criteria, which should simplify this greatly rather than the decorate-combine-undecorate pattern you've used. Try this:

```groovy
ASHLAR.out.tif
    .cross(ch_masks) { it[0]['id'] }
    .map{ t_ashlar, t_mask -> [t_mask[0], t_ashlar[1], t_mask[1]] }
    .combine(ch_mcquant_markers)
    .multiMap{ it ->
            image: [it[0], it[1]]
            mask: [it[0], it[2]]
            markers: [it[0], it[3]]
    }
    | MCQUANT
```

I'm not 100% sure I understand the behavior of cross, so this might not actually work...

Reply:
Wow, nice! Of course, cross is perfect here.  This is so much cleaner.
Seems to work great.  Tests are passing.